reward/staking_manager.go

```diff
@@ -132,12 +126,10 @@ func (sm *StakingManager) handleChainHeadEvent() {
 		// Handle ChainHeadEvent
 		case ev := <-sm.chainHeadChan:
 			if sm.governanceHelper.ProposerPolicy() == params.WeightedRandom {
-				stakingBlockNum := ev.Block.NumberU64() - ev.Block.NumberU64()%sm.governanceHelper.StakingUpdateInterval()
-				if cachedStakingInfo := sm.stakingInfoCache.get(stakingBlockNum); cachedStakingInfo == nil {
-					stakingInfo, _ := sm.updateStakingInfo(stakingBlockNum)
-					if stakingInfo == nil {
-						logger.Error("Failed to update stakingInfoCache", "blockNumber", ev.Block.NumberU64(), "stakingNumber", stakingBlockNum)
-					}
+				// check and update if staking info is not valid beforehand
+				stakingManager := sm.GetStakingInfo(ev.Block.NumberU64())
+				if stakingManager == nil {
+					logger.Error("unable to fetch staking info")
```

Comment:
Previous code tries to access/calculate  next staking info # `blocknumber - blocknumber % interval`.
But new code tries to access/calculate last staking info # `blocknumber  - blocknumber % interval - interval`.
We need to keep the logic to access/calculate next stakingInfo, not last one.

Reply:
`CalcStakingBlockNumber` is doing the calculation. `CalcStakingBlockNumber` is called in `GetStakingInfo`

Reply:
You were right. The code changes the logic. I've changed to fetch the previous code.